src/components/MySpace/Tools/index.tsx

```diff
@@ -41,19 +43,24 @@ function UserTools() {
   };
 
   const renderBody = () => {
-    if (isUserToolsLoading) {
+    if (isLoading) {
       return <Spinner size="sm" />;
     }
 
-    if (!isDeveloper) {
-      return <NoDeveloperAccess />;
+    if (!isLoading && !isDeveloper) {
+      if (!isUserRequestLoading && userRequest && userRequest?.length > 0) {
+        if (Object.keys(userRequest[userRequest?.length - 1]?.status)[0] !== "approved") {
+          return <NoDeveloperAccess access={Object.keys(userRequest[userRequest?.length - 1]?.status)[0]} />
```

Comment:
why do we need Object.keys here?

Reply:
![Screenshot 2024-11-29 at 12 51 33 PM](https://github.com/user-attachments/assets/d7741c22-9ead-45bf-a8b8-7db263b7fda1)

Status in response is a key with a null value, forgot to ask you about this

Reply:
ah okay in future pr can you create a utility function to convert these back and forth

Reply:
https://elna.atlassian.net/browse/ELNA-280
Added to backlog